pallets/ddc-staking/src/lib.rs

```diff
@@ -737,9 +835,94 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Pay out all the stakers for a single era.
+		#[pallet::weight(100_000)]
+		pub fn payout_stakers(origin: OriginFor<T>, era: EraIndex) -> DispatchResult {
+			ensure_signed(origin)?;
+			let current_era = Self::current_era().ok_or(Error::<T>::DDCEraNotSet)?;
+
+			// Makes sure this era hasn't been paid out yet
+			ensure!(!Self::paid_eras(era), Error::<T>::DoubleSpendRewards);
+
+			// This should be adjusted based on the finality of validation
+			ensure!(current_era >= era + 2, Error::<T>::EraNotValidated);
+
+			PaidEras::<T>::insert(era, true);
+			Self::do_payout_stakers(era)
+		}
+
+		/// Set price per byte of the bucket traffic in smallest units of the currency.
+		///
+		/// The dispatch origin for this call must be _Root_.
+		#[pallet::weight(10_000)]
+		pub fn set_pricing(origin: OriginFor<T>, price_per_byte: u128) -> DispatchResult {
+			ensure_root(origin)?;
+			<Pricing<T>>::set(Some(price_per_byte));
+			Ok(())
+		}
 	}
 
 	impl<T: Config> Pallet<T> {
+		pub fn do_payout_stakers(era: EraIndex) -> DispatchResult {
+			// ToDo: check that validation is finalised for era
```

Comment:
Is it still actual?

Reply:
why not? :)

Reply:
Because it's going to be merged to `master`. Why do we leave todos?

Reply:
Validation algorithm is not final yet, so I have no clue about the logic I should use. Current implementation works fine with current validation